qiskit_ibm_runtime/ibm_backend.py

```diff
@@ -755,7 +755,7 @@ def _runtime_run(
         if session and not session.active:
             raise RuntimeError(f"The session {session.session_id} is closed.")
         session_id = session.session_id if session else None
-        start_session = session_id is None
+        start_session = session is not None and session_id is None
```

Comment:
While the code works, I think it'd be easier to follow if we just group all the session related testing inside an `if` state, e.g. 

```
if self._session:
    if not session.active:
        raise ..
    session_id = self._session.session_id
    start_session = session_id is None
else:
    session_id = None
    start_session = False
```